syntax = "proto3";

package output_proto;

message OutputData {
    string rqId = 1;
    string model = 2;
    string market = 3;
    string currency = 4;
    double total_cost = 5;
    double obj = 6;
    double commission = 7;
    map<string, int32> lots_number = 8;
    map<string, int32> quantity = 9;
    map<string, int32>  weights = 10;
    map<string, int32> prices = 11;
    enum Statuses {
        NOT_STARTED = 0;
        STARTED = 1;
        OPTIMAL_FOUND = 2;
        INCORRECT_PROBLEM = 3; 
        UNBOUNDED_PROBLEM = 4;
        ITERATION_LIMIT_REACHED = 5;
        NODE_LIMIT_REACHED = 6;
        TIME_LIMIT_REACHED = 7;
        INTERRUPTED = 8;
        NUMERICAL_ERROR = 9;
        SUBOPTIMAL_FOUND = 10;
        USER_OBJ_LIMIT_REACHED = 11;
        NOT_SUPPORTED = 12;
    }
    Statuses status = 12;
    int32 iteration_amount = 13;
    double accuracy = 14;
}